examples/C++/DDS/HelloWorldExample/HelloWorldPublisher.cpp

```diff
@@ -39,13 +49,33 @@ HelloWorldPublisher::HelloWorldPublisher()
 {
 }
 
-bool HelloWorldPublisher::init()
+bool HelloWorldPublisher::init(
+        const std::string& topic_name,
+        uint32_t domain,
+        bool async,
+        const std::string& transport,
+        bool reliable,
+        bool transient)
 {
     hello_.index(0);
     hello_.message("HelloWorld");
```

Comment:
Regarding this topic, I would say it would be very good to have this information in the `message` so the Subscriber get it as user data.
It would needed to use another Publisher `std::string` created in `init` that is set regarding the Publisher attributes, and so the Subscriber will know which attributes the Publisher has.

Reply:
As discussed, we will include this request in the future if other arguments such as ownership are considered.